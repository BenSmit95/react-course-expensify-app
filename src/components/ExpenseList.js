import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {  
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />;
                })
            )
        }
    </div>
);

// Convention to name this function mapStateToProps
const mapStateToProps = (state) => {
    // return the props to the component
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
