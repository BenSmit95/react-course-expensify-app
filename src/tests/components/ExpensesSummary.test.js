import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary component correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={500} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with expenses data', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={500000} />);
    expect(wrapper).toMatchSnapshot();
});