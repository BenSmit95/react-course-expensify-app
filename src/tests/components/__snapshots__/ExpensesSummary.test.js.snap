// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary component correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $5.00
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with expenses data 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $5,000.00
  </h1>
</div>
`;
